### PR TITLE
WIP:  PdfBase::status method - tool for debugging

### PR DESCRIPTION
**Pull request by danielsibemol**
This PR includes the PdfBase::status method, a helpful tool in debugging. In addition, it contains a fix for a bug in incrementing indices in the ConvolutionPdf

### CODE REVIEW

File: python/PDFs/basic/CMakeLists.txt [hunk 1]
```diff
@@ -23,7 +23,7 @@ add_library(
   VoigtianPdf.cpp)
 
 target_link_libraries(_Basic PRIVATE _goofit_python)
-set_target_properties(_Basic PROPERTIES FOLDER python/PDFs)
+set_target_properties(_Basic PROPERTIES FOLDER python/PDFs CUDA_RESOLVE_DEVICE_SYMBOLS ON)
```
Comment (henryiii) [inline python/PDFs/basic/CMakeLists.txt:26]: This still bothers me - we shouldn't be forcing the device unconditionally, we need to be able to compile for arbitrary devices (especially such as in our testing CI runs!)

File: include/goofit/PdfBase.h [hunk 2]
```diff
@@ -120,6 +120,10 @@ class PdfBase {
     void registerFunction(std::string name, const T &function) {
         reflex_name_  = name;
         function_ptr_ = get_device_symbol_address(function);
+        // add the following 211222 to keep track of the correspondence
+        // between function pointer values and their names to make
+        // debugging/tracking the executionn of the code easier.  mds
```
Comment (henryiii) [inline include/goofit/PdfBase.h:125]: No need to add MDS's to everything, just adds clutter, and you can attribute the git commit to @mdsokoloff instead as a coauthor.

File: python/PDFs/basic/CMakeLists.txt (hunk 1 above)
Comment (danielsibemol) [reply python/PDFs/basic/CMakeLists.txt:26]: this has been added in a PR by Florian: https://github.com/GooFit/GooFit/pull/272. I think it was needed to use python bindings in with CUDA 11. Maybe we can remove this from this commit. @mdsokoloff, do you know if this was needed for your tests?

File: include/goofit/PdfBase.h (hunk 2 above)
Comment (danielsibemol) [reply include/goofit/PdfBase.h:125]: Mike is already the author of the first commit. In fact, these comments are directly from him, I just did not change it. I will clean these comments in my next push.

**Pull request merged**